cloud/pkg/leaderelection/leaderelection.go

```diff
@@ -69,6 +69,12 @@ func Run(cfg *config.CloudCoreConfig, readyzAdaptor *ReadyzAdaptor) {
 			// TODO: is it necessary to terminate the program gracefully?
 			//klog.Fatalf("leaderelection lost, rudely terminate program")
 			klog.Errorf("leaderelection lost, gracefully terminate program")
+			// Patch PodReadinessGate if program run in pod
+			err := TryToPatchPodReadinessGate(corev1.ConditionFalse)
```

Comment:
also try to reset podReadinessGate to false at the begin of Run, if cloudcore crashes and can not trigger OnStopedLeading last lifecycle
```go
func Run(cfg *config.CloudCoreConfig, readyzAdaptor *ReadyzAdaptor) {
    TryToPatchPodReadinessGate(corev1.ConditionFalse)
...
```

Thx for your outstanding contributing : )

Reply:
yes, commit updated.